src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +131,132 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+                cardNonceResponseReceived: (errors: NonceGenerationError[] | undefined, nonce: string | undefined, cardData: CardData | undefined,
+                                            billingContact: Contact | undefined, shippingContact: Contact | undefined) => {
+                    if (errors) {
+                        this._handleNonceGenerationErrors(errors);
+                    }
+
+                    if (nonce) {
+                        if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
+                            this._setExternalCheckoutData(cardData, nonce)
+                            .then(() => {
```

Comment:
This should be indented one extra level.

Also, with promises, you can just return the value and chained them on the outer most level:

```js
this._setExternalCheckoutData(cardData, nonce)
	.then(() => this._paymentInstrumentSelected(methodId))
	.then(() => squareOptions.onPaymentSelect && squareOptions.onPaymentSelect());
```

Reply:
👍